definitions/checks/foreman/check_https_proxies.rb

```diff
@@ -0,0 +1,35 @@
+module Checks
+  class CheckHttpsProxies < ForemanMaintain::Check
+    metadata do
+      label :https_proxies
+      for_feature :foreman_database
+      description 'Check for HTTPS proxies from DB'
+      manual_detection
+    end
+
+    def run
+      https_proxies = find_https_proxies
+      unless https_proxies.empty?
+        https_proxy_names = https_proxies.map { |proxy| proxy['name'] }
+        question = "Syncing repositories through an \'HTTP Proxy\' that uses the https:// "\
+                    "protocol is not supported directly with Satellite 6.10.\n"\
+                    "The following proxies use HTTPS: #{https_proxy_names.join(',')}.\n"\
+                    "For a workaround see https://access.redhat.com/solutions/6414991\n"\
+                    'Continue upgrade?'
+        answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
+        abort! if answer != :yes
```

Comment:
The whole pre_upgrade steps are running in fail_slow strategy so the abort does not help here. As it wont actually abort the upgrade instead it will just show [ABORTED] as status for the check. 
I like to know if aborting upgrade is required if https proxies are there? I think warning is enough? and if warning is enough then do we need question for the user?

Reply:
When I tested it the upgrade aborted after all of the steps finished running.  The abort is necessary because users with HTTPS proxies will be unable to sync any content until either the proxies are removed or the steps in the KCS article are accomplished

```

The processing was aborted by user during the following steps:

  [https-proxies]
```